raw_bsd.go

```diff
@@ -192,6 +200,73 @@ func (p *packetConn) SetPromiscuous(b bool) error {
 	return syscall.SetBpfPromisc(p.fd, m)
 }
 
+// ifReq is the BSD struct passed to ioctl for multicast add/drop operations.
+type ifReq struct {
+	ifrName [unix.IFNAMSIZ]uint8
+	ifrAddr unix.SockaddrDatalink
+}
+
+const etherAdderAlen = 6
+
+// JoinMulticast joins or drops membership of data link layer multicast address on interface,
+// allowing it to recieve multicast traffic.
+func (p *packetConn) SetMulticast(b bool, addr net.HardwareAddr) error {
```

Comment:
Will want to add a bounds check to check if `len(addr) == 6` for an Ethernet address I'd think. I don't know if this mechanism works with other BSD interface types that might have different address lengths. Any idea?